partiql-parser/src/main/kotlin/org/partiql/parser/V1PartiQLParser.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *      http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package org.partiql.parser
+
+import org.partiql.ast.v1.Statement
+import org.partiql.parser.internal.V1PartiQLParserDefault
+
+public interface V1PartiQLParser {
```

Comment:
Duplicated all of the existing `PartiQLParser` APIs for now to keep the other tests passing (e.g. planner, eval). After those other packages are migrated to use the parser outputting the new v1 AST, then we can delete the existing sprout AST `PartiQLParser` APIs.

Reply:
Yeah good catch and thanks for cutting the issue. Parser APIs should also be migrated to Java (especially the nested `Result` data class).

I'll link the issue in the kdoc.